src/Migration/Sources/CSV.php

```diff
@@ -255,9 +258,6 @@ private function exportDocuments(int $batchSize): void
 
                 $documentId = $parsedData['$id'] ?? 'unique()';
 
-                // `$id`, `$permissions` in the doc can cause issues!
-                unset($parsedData['$id'], $parsedData['$permissions']);
```

Comment:
I think the reason this was unset was probably due to the Document including the `$id` along with the data including the `$id`. Does this work fine without removing the `$id` from the parsed data?